plugins/scaffolder/src/components/fields/RepoUrlPicker/RepoUrlPicker.test.tsx

```diff
@@ -323,31 +324,33 @@ describe('RepoUrlPicker', () => {
       });
     });
 
-    it('should not call the scmAuthApi if secret is available in the state', async () => {
```

Comment:
I believe this test is no longer applicable as there is no way to guarantee the existing secret is for the host selected by this instance of the `RepoUrlPicker`. If we wish to keep this behavior, I believe we would need to additionally track the relationship of which host the credentials are for, which would bloat the SecretsContext and cause additional confusion.

Reply:
Yeah most probably.

I'd like to see a test however that the `scmAuthApi` is called with the different hosts as it looks like the test right now just checks the call after the last change?

Let's check that it calls the `scmAuthApi` with the correct hosts when changing and then that the latest secret value is the return from the last call with the correct host as a regression test.

Reply:
Changes applied in the latest commit